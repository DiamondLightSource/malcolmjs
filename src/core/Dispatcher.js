/*
 * React.js Starter Kit
 * Copyright (c) 2014 Konstantin Tarkus (@koistya), KriaSoft LLC.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.txt file in the root directory of this source tree.
 */

import Flux from 'flux';
import PayloadSources from '../constants/PayloadSources';
import assign from 'react/lib/Object.assign';

/**
 * A singleton that operates as the central hub for application updates.
 * For more information visit https://facebook.github.io/flux/
 */
let Dispatcher = assign(new Flux.Dispatcher(), {

  /**
   * @param {object} action The details of the action, including the action's
   * type and additional data coming from the server.
   */
  handleServerAction(action) {
    var payload = {
      source: PayloadSources.SERVER_ACTION,
      action: action
    };
    this.dispatch(payload);
  },

  /**
   * @param {object} action The details of the action, including the action's
   * type and additional data coming from the view.
   */
  handleViewAction(action) {
    var payload = {
      source: PayloadSources.VIEW_ACTION,
      action: action
    };
    this.dispatch(payload);
  }

});

export default Dispatcher;
