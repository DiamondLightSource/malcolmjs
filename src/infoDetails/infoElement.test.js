import React from 'react';
import { createShallow, createMount } from '@material-ui/core/test-utils';
import InfoElement from './infoElement.component';

describe('InfoElement', () => {
  let shallow;
  let mount;
  beforeEach(() => {
    shallow = createShallow();
    mount = createMount();
  });

  it('renders correctly', () => {
    const wrapper = shallow(
      <InfoElement
        key={1}
        label="Test element"
        value="testing..."
        alarm={null}
        tag="info:multiline"
      />
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });

  it('button hooks up correctly', () => {
    const clickHandler = jest.fn();
    const wrapper = mount(
      <InfoElement
        key={1}
        label="Test element"
        value={{ buttonLabel: 'test button', disabled: false }}
        alarm={null}
        tag="info:button"
        handlers={{ clickHandler }}
      />
    );
    wrapper
      .find('button')
      .last()
      .simulate('click');
    expect(clickHandler).toHaveBeenCalledTimes(1);
  });

  it('button disables correctly', () => {
    const clickHandler = jest.fn();
    const wrapper = mount(
      <InfoElement
        key={1}
        label="Test element"
        value={{ buttonLabel: 'test button', disabled: true }}
        alarm={null}
        tag="info:button"
        clickHandler={clickHandler}
      />
    );
    wrapper
      .find('button')
      .last()
      .simulate('click');
    expect(clickHandler).toHaveBeenCalledTimes(0);
  });
});
