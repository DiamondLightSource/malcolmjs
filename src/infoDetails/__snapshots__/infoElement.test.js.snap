// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoElement renders correctly 1`] = `
<div
  className="InfoElement-div-1"
>
  <WithStyles(Tooltip)
    id="1"
    placement="bottom"
    title=""
  >
    <WithStyles(IconButton)
      className="InfoElement-button-4"
      disableRipple={true}
    >
      <WithTheme(AttributeAlarm)
        alarmSeverity={null}
      />
    </WithStyles(IconButton)>
  </WithStyles(Tooltip)>
  <WithStyles(Typography)
    className="InfoElement-textName-2"
  >
    Test element
    : 
  </WithStyles(Typography)>
  <div
    className="InfoElement-controlContainer-3"
  >
    <WithStyles(WidgetTextUpdate)
      Text="testing..."
      noWrap={false}
    />
  </div>
</div>
`;
